engine/orchestration/src/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1353,6 +1363,20 @@ public void prepare(VirtualMachineProfile vm, DeployDestination dest) throws Sto
             disk.setDetails(getDetails(volumeInfo, dataStore));
 
             vm.addDisk(disk);
+
+            // If hypervisor is vSphere, check for clone type setting.
+            if (vm.getHypervisorType().equals(HypervisorType.VMware)) {
```

Comment:
So the idea is, instead of creating a clone which is linked to a parent, we would create a full clone of the disk. I can see this being useful when you don't want to have the overhead of nested reads/writes. Am I correct?

So there are reasons to use a `linked` clone as pointed out by @serg38  and there are reasons to use a full clone as well.